File: python/99_bottles.py
def bottle_song(n):
  for i in range(n, 0, -1):
    print(current_beer_count(i))
    print(take_one_down(i))
  print(current_beer_count(0))
  print(buy_more_beer(n))


def current_beer_count(n):
  return(f'{"No more" if n == 0 else n} bottle{"" if n == 1 else "s"} of beer on the wall, {"no more" if n == 0 else n} bottle{"" if n == 1 else "s"} of beer.')


def take_one_down(n):
  return(f'Take one down and pass it around, {"no more" if n == 1 else n - 1} bottle{"" if n - 1 == 1 else "s" } of beer on the wall.')


def buy_more_beer(n):
  return(f'Go to the store and buy some more, {n} bottles of beer on the wall.')


bottle_song(5)

File: js/99_bottles.js
// Run this file with node
// node 99_bottles.js [number of bottles] (defaults to 9)

function bottleSong(numBottles = 9) {
  singLine(numBottles);
  console.log(
    `No more bottles of beer on the wall, no more bottles of beer.
Go to the store and buy some more, ${numBottles} bottles of beer on the wall.`
  );
}

function singLine(numBottles) {
  if (numBottles > 0) {
    console.log(
      `${pluralBottles(numBottles)} of beer on the wall, ${pluralBottles(
        numBottles
      )} of beer.
Take one down and pass it around, ${pluralBottles(
        numBottles - 1
      ).toLowerCase()} of beer on the wall.`
    );
    singLine(numBottles - 1);
  }
}

function pluralBottles(num) {
  // Most common occurence
  if (num > 1) {
    return `${num} bottles`;
  }
  // Least common
  if (num === 1) {
    return "1 bottle";
  }
  // Last but not least...
  return "No more bottles";
}

bottleSong(process.argv[2]);
